OpenRA.Mods.Common/Traits/TooltipDescription.cs

```diff
@@ -20,15 +20,35 @@ public class TooltipDescriptionInfo : ConditionalTraitInfo
 		[Translate]
 		public readonly string Description = "";
 
-		public override object Create(ActorInitializer init) { return new TooltipDescription(this); }
+		[Desc("Player stances who can view the description.")]
+		public readonly Stance ValidStances = Stance.Ally | Stance.Neutral | Stance.Enemy;
+
+		public override object Create(ActorInitializer init) { return new TooltipDescription(init.Self, this); }
 	}
 
 	public class TooltipDescription : ConditionalTrait<TooltipDescriptionInfo>, IProvideTooltipInfo
 	{
-		public TooltipDescription(TooltipDescriptionInfo info)
-			: base(info) { }
+		readonly Actor self;
+		readonly TooltipDescriptionInfo info;
```

Comment:
Pretty sure you don't need this variable - ConditionalTrait already has the Info variable defined.